cql3/statements/select_statement.cc

```diff
@@ -301,42 +301,39 @@ select_statement::make_partition_slice(const query_options& options) const
         ++_stats.reverse_queries;
     }
 
-    const uint64_t per_partition_limit = get_inner_loop_limit(get_limit(options, _per_partition_limit),
+    const uint64_t per_partition_limit = get_inner_loop_limit(get_limit(options, _per_partition_limit, true),
         _selection->is_aggregate());
     return query::partition_slice(std::move(bounds),
         std::move(static_columns), std::move(regular_columns), _opts, nullptr, per_partition_limit);
 }
 
-select_statement::get_limit_result select_statement::get_limit(
-    const query_options& options, const std::optional<expr::expression>& limit, bool is_per_partition_limit) const
+uint64_t select_statement::get_limit(const query_options& options, const std::optional<expr::expression>& limit, bool is_per_partition_limit) const
```

Comment:
Why the change from result to return+exceptions?

Reply:
Mixing such changes into that which is the subject of this PR is distracting and makes the PR harder to review.